owncloudApp/src/main/java/com/owncloud/android/NetworkBoundResource.kt

```diff
@@ -39,82 +39,88 @@ import androidx.annotation.MainThread
 import androidx.annotation.WorkerThread
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MediatorLiveData
-import androidx.lifecycle.MutableLiveData
 import com.owncloud.android.lib.common.operations.RemoteOperationResult
 import com.owncloud.android.vo.Resource
 
 abstract class NetworkBoundResource<ResultType, RequestType>(
     private val appExecutors: AppExecutors
 ) {
-
     /**
-     * Result will observe three different livedata objects and react on change events from them
+     * A generic class that can provide a resource backed by both the sqlite database and the network.
      * - Shares livedata from Room to detect changes in database
      * - Errors livedata from remote operations
      * - Loading status
      */
     private val result = MediatorLiveData<Resource<ResultType>>()
 
     init {
-        val dbSource = loadFromDb()
-
-        result.addSource(dbSource) { data ->
-            result.value = Resource.success(data)
-        }
-
-        fetchFromNetwork()
+        @Suppress("LeakingThis")
```

Comment:
just curious: what was Android Studio complaining about??

Reply:
Because I'm calling non-final function (loadFromDb) in constructor